client/src/holdings/AddTradeDialog.tsx

```diff
@@ -48,101 +122,111 @@ export function AddTradeDialog({
-          }}>
-          {({isSubmitting}) => (
-            <Form>
-              <DialogTitle id='form-dialog-title'>
-                Add Trade ({holding.symbol})
-              </DialogTitle>
-              <DialogContent className='DialogContent'>
-                <MuiPickersUtilsProvider utils={DateFnsUtils}>
-                  <Field
-                    component={KeyboardDateTimePicker}
-                    label='Date'
-                    variant='inline'
-                    name='date'
-                  />
-                </MuiPickersUtilsProvider>
-                <Field as='select' name='category'>
-                  <option value='BUY'>Buy</option>
-                  <option value='SELL'>Sell</option>
-                </Field>
-                <Field component={TextField} label='Broker' name='broker' />
-                <Field component={TextField} label='Quantity' name='quantity' />
-                <Field as='select' name='priceCurrency'>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <DialogTitle id='form-dialog-title'>
+            Add Trade ({holding.symbol})
+          </DialogTitle>
+          <DialogContent className='DialogContent'>
+            <div className='Category'>
+              <Controller
+                render={(category) => (
+                  <ToggleButtonGroup
+                    exclusive
+                    size={'medium'}
+                    aria-label='Category'
+                    {...category}
+                    onChange={(e, value) => {
+                      category.onChange(value);
+                    }}>
+                    <ToggleButton value='BUY' key='BUY'>
+                      BUY
+                    </ToggleButton>
+                    <ToggleButton value='SELL' key='SELL'>
+                      SELL
+                    </ToggleButton>
+                  </ToggleButtonGroup>
+                )}
+                label='Category'
+                name='category'
+                control={control}
+                inputRef={register}
+              />
+            </div>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <DateTimePicker
+                value={selectedDate}
+                onChange={handleDateChange}
+                label='Date'
+                variant='inline'
+                name='date'
+                inputRef={register}
+              />
+            </MuiPickersUtilsProvider>
+            <TextField
+              margin='dense'
+              label='Broker'
+              name='broker'
+              inputRef={register}
+            />
+            <TextField
+              margin='dense'
+              label='Quantity'
+              name='quantity'
+              inputRef={register}
+              error={errors.quantity ? true : false}
+            />
+            <Controller
+              name='priceCurrency'
+              label='Currency'
+              control={control}
+              as={
```

Comment:
Can you use the render method rather than 'as', because 'as' is being deprecated soon: https://react-hook-form.com/api#Controller

Reply:
@willbeaufoy - done but you might need to take a look as I couldn't get the default value to display without passing it again at the Select component level - not sure why.